packages/daimo-userop/src/daimoOpSender.ts

```diff
@@ -155,26 +165,65 @@ export class DaimoOpSender {
     return this.sendUserOp(op);
   }
 
-  /** Creates an ephemeral note with given value. Returns userOpHash. */
+  /** Creates an ephemeral note V2 with given value. Returns userOpHash. */
   public async createEphemeralNote(
     ephemeralOwner: `0x${string}`,
     amount: `${number}`,
+    approveFirst: boolean,
     opMetadata: DaimoOpMetadata
   ) {
-    const { tokenDecimals, notesAddress } = this.opConfig;
+    const { tokenDecimals, tokenAddress, notesAddressV2 } = this.opConfig;
 
     const parsedAmount = parseUnits(amount, tokenDecimals);
     console.log(`[OP] create ${parsedAmount} note for ${ephemeralOwner}`);
 
+    const executions = [
+      {
+        dest: notesAddressV2,
+        value: 0n,
+        data: encodeFunctionData({
+          abi: Contracts.daimoEphemeralNotesV2ABI,
+          functionName: "createNote",
+          args: [ephemeralOwner, parsedAmount],
+        }),
+      },
+    ];
+
+    if (approveFirst) {
+      executions.unshift({
+        // Approve notes contract infinite spending on behalf of the account
+        dest: tokenAddress,
+        value: 0n,
+        data: encodeFunctionData({
+          abi: erc20ABI,
+          functionName: "approve",
+          args: [notesAddressV2, maxUint256],
+        }),
+      });
+    }
+
+    const op = this.opBuilder.executeBatch(executions, opMetadata);
+
+    return this.sendUserOp(op);
+  }
+
+  /** Claims an ephemeral note. Returns userOpHash. */
+  public async claimEphemeralNoteV1(
+    ephemeralOwner: `0x${string}`,
+    signature: `0x${string}`,
+    opMetadata: DaimoOpMetadata
+  ) {
+    console.log(`[OP] claim ephemeral note ${ephemeralOwner}`);
+
     const op = this.opBuilder.executeBatch(
       [
         {
-          dest: notesAddress,
+          dest: this.opConfig.notesAddress,
           value: 0n,
           data: encodeFunctionData({
             abi: Contracts.daimoEphemeralNotesABI,
-            functionName: "createNote",
-            args: [ephemeralOwner, parsedAmount],
+            functionName: "claimNote",
+            args: [ephemeralOwner, signature],
```

Comment:
wait how were we able to pass `parsedAmount` instead of `signature` before? this is for the existing notes contract

Reply:
git diff just looks weird here, the `parsedAmount` stuff is from `createNote`, not claim